src/pages/HistoryPage.js

```diff
@@ -10,32 +10,88 @@ const HistoryPage = props => {
   const [transactions, setTransactions] = useState([]);
 
   if (!gotTransactions) {
-    props.firebase
-      .transactions()
-      .get()
-      .then(querySnapshot => {
-        const transacs = querySnapshot.docs.map(doc => {
-          return {id: doc.id, ...doc.data()};
-        });
-        transacs.forEach(transac => {
-          props.firebase
-            .user(
-              transac.providerID === props.firebase.getMyUID()
-                ? transac.consumerID
-                : transac.providerID
+    console.log('////', props);
+
+    props.firebase.myUID &&
+      props.firebase
+        .user(props.firebase.getMyUID())
+        .get()
+        .then(user => {
+          console.log('user12///', user.data());
+          Promise.all(
+            user.data().transactions.map(id =>
```

Comment:
I got error on this line. 

"Unhandled Rejection (TypeError): Cannot read property 'map' of undefined"

Reply:
I will investigate this.